course_discovery/apps/course_metadata/management/commands/backfill_history.py

```diff
@@ -0,0 +1,173 @@
+            history_date = input_filename.rsplit('_')[-1]
+
+            with connection.cursor() as cursor:
+                query = u"""
+                    SELECT
+                        column_name
+                    FROM information_schema.columns
+                    WHERE table_name='{}'
+                    ORDER BY ordinal_position
+                    """.format(table)
+                cursor.execute(query)
+                columns = [column[0] for column in cursor.fetchall()]
+
+            # We don't record the AutoSlugField fields in the historical model.
+            for column_name in ['slug', 'url_slug']:
+                if column_name in columns:
+                    columns.remove(column_name)
+
+            with open(file_path, 'r') as input_file:
+                reader = csv.DictReader(input_file, delimiter='\x01')
+                lines = list(reader)
+
+                for rows in self.chunks(lines, batch_size):
+                    row_ids = [row['ID'] for row in rows]
+                    if table == 'courses_course':
+                        ids = ','.join("'{}'".format(id) for id in row_ids)
+                    else:
+                        ids = ','.join(row_ids)
+
+                    # Checks for existing historical records
+                    with connection.cursor() as cursor:
+                        query = u"""
+                            SELECT COUNT(1)
+                            FROM {historical_table}
+                            WHERE ID in ({ids})
+                            AND history_type='+'
+                            """.format(
+                                historical_table=historical_table,
+                                ids=ids
+                                )  # noqa
+                        cursor.execute(query)
+                        count = cursor.fetchone()[0]
+
+                    # pylint: disable=no-else-continue
+                    if count == len(rows):
+                        log.info(
+                            u"Initial history records already exist for ids %s..%s - skipping.",
+                            ','.join(row_ids[:2]), ','.join(row_ids[-2:])
+                        )
+                        continue
+                    elif count != 0:
+                        raise Exception(u"Database count: %s does not match input count: %s" % (count, len(rows)))
+
+                    values = [[row[column.upper()] for column in columns] for row in rows]
+
+                    # Replace 'NULL' with None
+                    values = self.replace_values(values, 'NULL', None)
+                    # Replace 'true' with True
+                    values = self.replace_values(values, 'true', True)
+                    # Replace 'false' with False
+                    values = self.replace_values(values, 'false', False)
+                    # Add history columns data
+                    for value in values:
```

Comment:
I know this is just a one-off command, but I'll just say it anyway:
```suggestion
                    for row in values:
```